kernel/src/schema/mod.rs

```diff
@@ -212,33 +216,107 @@ impl StructField {
             .collect()
     }
 
-    /// Applies physical name mappings to this field
+    /// Applies physical name mappings to this field. If the `column_mapping_mode` is
+    /// [`ColumnMappingMode::Id`], then each StructField will have its parquet field id in the
+    /// [`ColumnMetadataKey::ParquetFieldId`] metadata field.
+    ///
     ///
     /// NOTE: Caller affirms that the schema was already validated by
     /// [`crate::table_features::validate_schema_column_mapping`], to ensure that annotations are
     /// always and only present when column mapping mode is enabled.
```

Comment:
noticed that we previously had this requirement - if we pass column mapping down here now perhaps it would make sense to push down this into the method and remove this requirement?

Reply:
The problem is that `make_physical` is typically used as a`StructField` in StateInfo::try_new(). Validating column mapping repeatedly for each StructFiled would likely be rly expensive.